### PR TITLE
Remove overlayscrollbar

### PR DESCRIPTION
**Pull request by 00drip**
Refer to #44 

Features:
**layout**: replace overlay-scrollbars with simplebar (04b46d4)

### CODE REVIEW

File: pages/chat/_entity.vue [hunk 1]
```diff
@@ -23,25 +23,26 @@
...
+      <div class="d-flex flex-column flex-grow-1">
         <!-- Message List -->
-        <overlay-scrollbars
+        <simplebar
+          class="simplebar wide-scrollbar flex-grow-1" 
+          data-simplebar-auto-hide="false" 
+          data-simplebar-force-visible="true"
           ref="messageList"
-          class="flex-grow-1 flex-shrink-1 wide-scrollbar"
           :options="{scrollbars:{clickScrolling: true}}">
```
Comment (misterupkeep) [inline pages/chat/_entity.vue:38]: The `options` attribute does nothing to `<simplebar/>`  components.
```suggestion
          >
```
By the way, does click scrolling work on simplebar (and how)?

File: components/SideBar.vue [hunk 2]
```diff
@@ -1,13 +1,14 @@
 <template>
   <div class="sidebar grey-100 d-flex flex-column">
     <header-bar pad-bottom/>
-    <overlay-scrollbars class="narrow-scrollbar flex-grow-1"
+    <simplebar class="simplebar narrow-scrollbar flex-grow-1" style="min-height: 0;"
+      data-simplebar-auto-hide="false" data-simplebar-force-visible="true"
```
Comment (misterupkeep) [inline components/SideBar.vue:5]: The scrollbar should autohide here (just like it did before on OS)
```suggestion
```

File: pages/chat/_entity.vue [hunk 3]
```diff
@@ -23,25 +23,26 @@
...
-      <div class="d-flex flex-column flex-grow-1 os-host-flexbox">
+      <div class="d-flex flex-column flex-grow-1">
         <!-- Message List -->
-        <overlay-scrollbars
+        <simplebar
+          class="simplebar wide-scrollbar flex-grow-1" 
+          data-simplebar-auto-hide="false" 
+          data-simplebar-force-visible="true"
           ref="messageList"
-          class="flex-grow-1 flex-shrink-1 wide-scrollbar"
```
Comment (misterupkeep) [inline pages/chat/_entity.vue:36]: Didn't `.wide-scrollbar` also draw a slightly transparent (but visible) track, too? I don't think `<simplebar/>` has that by default.

File: components/SideBar.vue [hunk 4]
```diff
@@ -1,13 +1,14 @@
 <template>
   <div class="sidebar grey-100 d-flex flex-column">
     <header-bar pad-bottom/>
-    <overlay-scrollbars class="narrow-scrollbar flex-grow-1"
+    <simplebar class="simplebar narrow-scrollbar flex-grow-1" style="min-height: 0;"
```
[the remark this answers is not in the document]
Comment (misterupkeep) [reply components/SideBar.vue:4]: I found it in `_entity.vue` inside a scoped style element. I know it's on a page component, but does it even do anything here? And why is it in there in the first place? You deleted the OS styles from `/assets/reset.scss`, why not add that in place, too?

File: pages/chat/_entity.vue [hunk 5]
```diff
@@ -70,6 +72,10 @@
     @include ensure-width(300px)
   }
 
+  .simplebar {
+    min-height: 0;
+  }
+
```
Comment (misterupkeep) [inline pages/chat/_entity.vue:78]: Why is this here, and not in `reset.scss`, where every other CSS reset is? I'm assuming this is a default simplebar class, and not something you have to add. Either way, it should be in a global CSS stylesheet (and not embedded here)

```suggestion
```

File: components/SideBar.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (misterupkeep) [reply components/SideBar.vue:4]: What's worse is that the `.simplebar` class applies the inline style here.
```suggestion
    <simplebar class="narrow-scrollbar flex-grow-1" style="min-height: 0;"
```
```suggestion
    <simplebar class="simplebar narrow-scrollbar flex-grow-1"
```

File: pages/settings/_submenu.vue [hunk 6]
```diff
@@ -39,8 +40,4 @@
 
 <style scoped lang="scss">
 .component { background: map-get($black, "lighten") !important; }
-
-// <overlay-scrollbars/> loses all its padding when destroyed, causing a visual jump
-// The solution: use margins. Unfortunately, one of its inner elements has a top padding, which we disable here.
-*::v-deep .os-content { padding-top: 0 !important; }
```
Comment (misterupkeep) [inline pages/settings/_submenu.vue:45]: Speaking of which, the spacings between content (and screen/layout edges) are all the same?

File: pages/settings/index.vue [hunk 7]
```diff
@@ -29,12 +29,13 @@
   </div>
 
   <div v-else class="lhs-bg w-100 h-100">
-    <overlay-scrollbars class="narrow-scrollbar h-100"
-                        :options="{scrollbars: {autoHide: 'leave', autoHideDelay: 0}}">
+    <simplebar class="simplebar narrow-scrollbar h-100"
+               data-simplebar-auto-hide="false" data-simplebar-force-visible="true"
+               :options="{scrollbars: {autoHide: 'leave', autoHideDelay: 0}}">
```
Comment (misterupkeep) [inline pages/settings/index.vue:34]: The `options` attribute does nothing on `<simplebar/>` components
```suggestion
               >
```

**Review by misterupkeep — CHANGES_REQUESTED** [review]
Has a fair bit of issues before it is as good as overlay-scrollbars was before this change

File: pages/chat/_entity.vue (hunk 1 above)
Comment (00drip) [reply pages/chat/_entity.vue:38]: >By the way, does click scrolling work on simplebar (and how)?

Yes, it's somewhat weird though. It doesn't snap to the destination but moves towards it in steps. There's no setting for this AFAIK so we'll probably have to make an issue/PR.

File: components/SideBar.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (00drip) [reply components/SideBar.vue:4]: Putting that in `reset.scss` doesn't break anything. I'll remove the inline style as well.

File: pages/settings/_submenu.vue (hunk 6 above)
Comment (00drip) [reply pages/settings/_submenu.vue:45]: I didn't measure anything but nothing looks too high or low. Even when the window is really small the scrollbar reaches the edges of the container as expected.

Comment (misterupkeep) [reply pages/settings/_submenu.vue:45]: What about on mobile?

File: pages/chat/_entity.vue (hunk 3 above)
Comment (00drip) [reply pages/chat/_entity.vue:36]: Added.

File: plugins/overlayscrollbars.js [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    nativeScrollbarsOverlaid: {
+        initialize: false
+    },
+    overflowBehavior: {
+        x: 'hidden',
+    },
+    scrollbars: {
+        clickScrolling: true,
+    },
+});
```
Comment (misterupkeep) [inline plugins/overlayscrollbars.js]: How is this file back?

File: pages/settings/index.vue [hunk 9]
```diff
@@ -29,9 +29,8 @@
   </div>
 
   <div v-else class="lhs-bg w-100 h-100">
-    <simplebar class="simplebar narrow-scrollbar h-100"
-               data-simplebar-auto-hide="false" data-simplebar-force-visible="true"
-               :options="{scrollbars: {autoHide: 'leave', autoHideDelay: 0}}">
+    <simplebar class="simplebar wide-scrollbar h-100"
+               data-simplebar-auto-hide="false" data-simplebar-force-visible="true">
```
Comment (misterupkeep) [inline pages/settings/index.vue:33]: This scrollbar still doesn't autohide

File: pages/chat/_entity.vue [hunk 10]
```diff
@@ -31,11 +30,10 @@
       <div class="d-flex flex-column flex-grow-1">
         <!-- Message List -->
         <simplebar
-          class="simplebar wide-scrollbar flex-grow-1" 
-          data-simplebar-auto-hide="false" 
+          class="simplebar simplebar-no-gutter wide-scrollbar flex-grow-1" 
+          data-simplebar-auto-hide="true"
```
Comment (misterupkeep) [inline pages/chat/_entity.vue:34]: Why does the message window (attempt to) auto hide?
```suggestion
          data-simplebar-auto-hide="false"
```

File: assets/reset.scss [hunk 11]
```diff
@@ -34,7 +34,48 @@
...
+    border-radius: .7em;
+  }
+}
 
 b {
   font-weight: 600 !important;
 }
+
+.simplebar {
+  min-height: 0;
```
Comment (misterupkeep) [inline assets/reset.scss:56]: If this should be the default (it's sprinkled on every `<simplebar/>`), why does it have to be explicitly added?

**Review by misterupkeep — CHANGES_REQUESTED** [review]
The `min-height: 0` `.simplebar` reset ought to be opt-out, since it's ubiquitously present anyway. And there's just one auto-hide that needs fixing.

**Comment by misterupkeep** [comment]
I'll mess around a bit with this myself in a few, and then I'll merge (and then probably cascade that merge into `master`)
